subsys/bluetooth/audio/capabilities.c

```diff
@@ -351,9 +332,18 @@ int bt_audio_capability_register(struct bt_audio_capability *cap)
 	static bool pacs_cb_registered;
 	sys_slist_t *lst;
 
-	if (!cap || !cap->codec) {
-		return -EINVAL;
-	}
+	__ASSERT(cap, "invalid parameters\n");
```

Comment:
Generally I would recommend using `CHECKIF` over `__ASSERT` or `if` as it can server as either (based on a Kconfig), or even be disabled if the application code is well tested. This gives the application developer more control.